pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -428,6 +463,29 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    logBrokerResponse(requestStatistics, requestId, query, compilationStartTimeNs, brokerRequest,
+        numUnavailableSegments, serverStats, brokerResponse, executionEndTimeNs);
+    return brokerResponse;
+  }
+
+  /** Given a {@link BrokerRequest}, check if the WHERE clause will always evaluate to false. */
+  private boolean isFilterAlwaysFalse(BrokerRequest brokerRequest) {
+    return brokerRequest == null || brokerRequest.getPinotQuery() == null || (
+        brokerRequest.getPinotQuery().getFilterExpression() != null && brokerRequest.getPinotQuery()
+            .getFilterExpression().equals(FALSE));
+  }
```

Comment:
Suggest handling the null check in the caller side, and simplify the method. Same for `isFilterAlwaysTrue`
```suggestion
  private boolean isFilterAlwaysFalse(BrokerRequest brokerRequest) {
    return FALSE.equals(brokerRequest.getPinotQuery().getFilterExpression());
  }
```

Reply:
Moved null check to caller, although I thought the previous way was better :-)

Reply:
Hmm, when brokerRequest is `null`, returning `true` for `isFilterAlwaysFalse()` but returning `false` for `isFilterAlwaysTrue()` is confusing. Basically these 2 method does not apply to `null` brokerRequest.

Reply:
Fixed.